contracts/src/v0.8/dev/VRFCoordinatorV2.sol

```diff
@@ -0,0 +1,435 @@
+    //
+    // Public key. Skips byte array's length prefix.
+    uint256 public constant PUBLIC_KEY_OFFSET = 0x20;
+    // Seed is 7th word in proof, plus word for length, (6+1)*0x20=0xe0
+    uint256 public constant PRESEED_OFFSET = 0xe0;
+
+    function fulfillRandomWords(
+        bytes memory _proof
+    )
+    external
+    {
+        uint256 startGas = gasleft();
+        (bytes32 keyHash, uint256 requestId,
+        uint256 randomness, FulfillmentParams memory fp) = getRandomnessFromProof(_proof);
+
+        uint256[] memory randomWords = new uint256[](fp.numWords);
+        for (uint256 i = 0; i < fp.numWords; i++) {
+            randomWords[i] = uint256(keccak256(abi.encode(randomness, i)));
+        }
+
+        // Prevent re-entrancy. The user callback cannot call fulfillRandomWords again
+        // with the same proof because this getRandomnessFromProof will revert because the requestId
+        // is gone.
+        delete s_callbacks[requestId];
+        bytes memory resp = abi.encodeWithSelector(FULFILL_RANDOM_WORDS_SELECTOR, requestId, randomWords);
+        require(gasleft() > fp.callbackGasLimit, "not enough gas for consumer");
+        (bool success,) = fp.sender.call(resp);
+        // Avoid unused-local-variable warning. (success is only present to prevent
+        // a warning that the return value of consumerContract.call is unused.)
+        (success);
+
+        emit RandomWordsFulfilled(requestId, randomWords, true);
+        // We want to charge users exactly for how much gas they use in their callback.
+        // The gasAfterPaymentCalculation is meant to cover these additional operations where we
+        // decrement the subscription balance and increment the oracles withdrawable balance.
+        uint256 payment = calculatePaymentAmount(startGas, s_config.gasAfterPaymentCalculation);
+        s_subscriptions[fp.subId].balance -= payment;
+        s_withdrawableTokens[s_serviceAgreements[keyHash]] += payment;
+    }
+
+    function calculatePaymentAmount(
+        uint256 startGas,
+        uint256 gasAfterPaymentCalculation
+    )
+    private
+    view
+    returns (uint256)
+    {
+        // Get the amount of gas used for (fulfillment + request)
+        uint256 gasWei; // wei/gas i.e. gasPrice
+        uint256 linkWei; // link/wei i.e. link price in wei.
+        (gasWei, linkWei) = getFeedData();
+        // (1e18 linkWei/link) (wei/gas * gas) / (wei/link) = linkWei
+        return 1e18*gasWei*(gasAfterPaymentCalculation + startGas - gasleft()) / linkWei;
+    }
+
+    function getRandomnessFromProof(bytes memory _proof)
+    public view returns (bytes32 currentKeyHash,
+        uint256 requestId, uint256 randomness, FulfillmentParams memory fp) {
+        // blockNum follows proof, which follows length word (only direct-number
+        // constants are allowed in assembly, so have to compute this in code)
+        uint256 BLOCKNUM_OFFSET = 0x20 + PROOF_LENGTH;
```

Comment:
Maybe make it a constant and then just pull it into a local variable to save the little computation? At the very least I'd recommend changing the constant style casing on a non-constant variable, it is unconventional and a little misleading.